pandas/core/dtypes/cast.py

```diff
@@ -1779,13 +1780,14 @@ def ensure_nanosecond_dtype(dtype: DtypeObj) -> DtypeObj:
     return dtype
 
 
-def find_common_type(types: list[DtypeObj]) -> DtypeObj:
+def find_common_type(types: list[DtypeObj], *, strict_uint64: bool = False) -> DtypeObj:
     """
     Find a common data type among the given dtypes.
 
     Parameters
     ----------
     types : list of dtypes
+    strict_uint64 : if True, object dtype is returned if uint64 and signed int present.
```

Comment:
why do we not *always* do this?

Reply:
FWIW ive determined that this is the fix for #41775, though the place for the fix is in Index._find_common_type_compat